pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,470 @@
+		requestType = ipv4.ICMPTypeEcho
+	}
+
+	timeStart := time.Now()
+	seqID := getICMPSeq()
+	body := &icmp.Echo{
+		ID:   icmpEchoID,
+		Seq:  int(seqID),
+		Data: []byte(timeStart.Format(time.RFC3339Nano)),
+	}
+	msg := icmp.Message{
+		Type: requestType,
+		Code: 0,
+		Body: body,
+	}
+	klog.V(4).InfoS("Sending ICMP message", "IP", ip, "SeqID", seqID, "body", body)
+
+	// Serialize the ICMP message
+	msgBytes, err := msg.Marshal(nil)
+	if err != nil {
+		return err
+	}
+
+	// Send the ICMP message
+	_, err = socket.WriteTo(msgBytes, ip)
+	if err != nil {
+		return err
+	}
+
+	// Create or update the latency store
+	mutator := func(entry *NodeIPLatencyEntry) {
+		entry.LastSendTime = timeStart
+	}
+	m.latencyStore.SetNodeIPLatencyEntry(addr.String(), mutator)
+
+	return nil
+}
+
+// recvPing receives an ICMP message from the target IP address.
+func (m *NodeLatencyMonitor) recvPing(socket net.PacketConn, isIPv4 bool) {
+	// We only expect small packets, if we receive a larger packet, we will drop the extra data.
+	readBuffer := make([]byte, 128)
+	for {
+		n, peer, err := socket.ReadFrom(readBuffer)
+		if err != nil {
+			// When the socket is closed in the Run method, this error will be logged, which is not ideal.
+			// In the future, we may try setting a ReadDeadline on the socket before each ReadFrom and using
+			// a channel to signal that the loop should terminate.
+			klog.ErrorS(err, "Failed to read ICMP message")
+			return
+		}
+
+		destIP := peer.String()
+
+		// Parse the ICMP message
+		var msg *icmp.Message
+		if isIPv4 {
+			msg, err = icmp.ParseMessage(ProtocolICMP, readBuffer[:n])
+			if err != nil {
+				klog.ErrorS(err, "Failed to parse ICMP message")
+				continue
+			}
+			if msg.Type != ipv4.ICMPTypeEcho && msg.Type != ipv4.ICMPTypeEchoReply {
+				klog.V(5).InfoS("Ignoring non-ping ICMP message", "msg", msg)
+				continue
+			}
+			// Ignore ICMP echo messages received from other Nodes (they will be answered by the system)
+			if msg.Type == ipv4.ICMPTypeEcho {
+				klog.V(7).InfoS("Ignoring ICMP echo request message", "msg", msg)
+				continue
+			}
+		} else {
+			msg, err = icmp.ParseMessage(ProtocolICMPv6, readBuffer)
+			if err != nil {
+				klog.ErrorS(err, "Failed to parse ICMP message")
+				continue
+			}
+			if msg.Type != ipv6.ICMPTypeEchoRequest && msg.Type != ipv6.ICMPTypeEchoReply {
+				klog.V(5).InfoS("Ignoring non-ping ICMP message", "msg", msg)
+				continue
+			}
+			// Ignore ICMP echo messages received from other Nodes (they will be answered by the system)
+			if msg.Type == ipv6.ICMPTypeEchoRequest {
+				klog.V(7).InfoS("Ignoring ICMP echo request message", "msg", msg)
+				continue
+			}
+		}
+
+		echo, ok := msg.Body.(*icmp.Echo)
+		if !ok {
+			klog.ErrorS(nil, "Failed to assert type as *icmp.Echo")
+			continue
+		}
+		if echo.ID != icmpEchoID {
+			klog.V(4).InfoS("Ignoring ICMP message with wrong echo ID", "msg", msg)
+		}
+
+		klog.V(4).InfoS("Recv ICMP message", "IP", destIP, "msg", msg)
```

Comment:
```suggestion
		klog.V(4).InfoS("Received ICMP message", "IP", destIP, "msg", msg)
```
Use past tense to avoid ambiguity.

Reply:
Thanks!